Framework/Core/src/ExternalFairMQDeviceProxy.cxx

```diff
@@ -498,11 +501,19 @@ InjectorFunction dplModelAdaptor(std::vector<OutputSpec> const& filterSpecs, DPL
       }
     } // end of loop over parts
 
+    bool didSendParts = false;
     for (auto& [channelName, channelParts] : outputs) {
       if (channelParts.Size() == 0) {
         continue;
       }
-      sendOnChannel(device, channelParts, channelName, newTimesliceId);
+      didSendParts = true;
```

Comment:
if we call the injectMissingInputs and it sends something, we also need to set this to true.

Reply:
That's already handled, no?

Reply:
Actually, wait, these lines are in the converter function, right?
But you call injectMissingData before you call the converter?
Doesn't that mean we will always inject, even if there is nothing in the current iteration?

Reply:
I think it should be fine, because in that case we do:

https://github.com/AliceO2Group/AliceO2/pull/12015/files#diff-85aaa9e35fe2274e483945e3b2742245e7206421d3837404f4bf6007d3978434L314-L318